### ISSUE TITLE
Rule Change: provide an option to turn off counting of optional chaining for complexity

### ISSUE THREAD

**Issue by vitalets** [eslint/eslint#18432]
### What rule do you want to change?

complexity

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
function a(x) {
   return a?.b?.c?.d;
}
```

### What does the rule currently do for this code?

Calcs complexity as 4

### What will the rule do after it's changed?

Calc complexity as 1 

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

After https://github.com/eslint/eslint/issues/18060 complexity takes into account optional chaining when calculating its value.
I'd like to request an option to disable optional chaining counting for complexity.
I agree with the original issue discussion, that formally optional chaining creates separate execution branches. 
But if we talk about *visual complexity* it's quite different. 

Compare these two examples:
(1)
```js
function a(x) {
   return a?.b?.c?.d;
}
```
(2)
```js
function a(x) {
  if (condition) {
    if (condition) {
        return a;
      } else if (condition) {
        return b;
     } else {
       return c;
     }
  } else {
    return d;
  }
}
```
I'm totally fine to have expression (1) in the codebase. But would not like to have expression (2) as it's really hard to read.

**Comment by mdjermanovic** [comment]
Hi @vitalets, thanks for the issue!

As stated in the [documentation](https://eslint.org/docs/latest/rules/complexity), this rule specifically calculates [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity). In terms of cyclomatic complexity, the two examples are equivalent because they have the same number of branches / code paths.

If you want to use a different metric, you can always copy the core rule into a custom rule and modify the calculations to exclude optional chaining.

**Comment by vitalets** [comment]
I've created a custom rule for calculating complexity without optional chaining:
https://github.com/vitalets/eslint-plugin-visual-complexity
